tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -191,18 +192,19 @@ public function filterResponseAccordingToNotificationSubject(string $subject): o
 	}
 
 	/**
-	 * @Then user :user should get a notification with subject :subject and message:
+	 * @Then user :user should get a notification with subject :subject, resource :resource and message:
 	 *
 	 * @param string $user
 	 * @param string $subject
+	 * @param string $resource
 	 * @param TableNode $table
 	 *
 	 * @return void
 	 */
-	public function userShouldGetANotificationWithMessage(string $user, string $subject, TableNode $table):void {
+	public function userShouldGetANotificationWithMessage(string $user, string $subject, string $resource, TableNode $table):void {
 		$this->userListAllNotifications($user);
 		$this->featureContext->theHTTPStatusCodeShouldBe(200);
-		$actualMessage = str_replace(["\r", "\n"], " ", $this->filterResponseAccordingToNotificationSubject($subject)->message);
+		$actualMessage = str_replace(["\r", "\n"], " ", $this->filterResponseAccordingToNotificationSubject($subject, $resource)->message);
 		$expectedMessage = $table->getColumnsHash()[0]['message'];
```

Comment:
Maybe we check it like this, we check for the notification for this subject and the we mark it as read ( delete it )so that the next step will get the actual notification.
You might need to implement `deleteNotification`
```suggestion
$notification = $this->filterResponseAccordingToNotificationSubject($subject);
		if (isset($notification->messageRichParameters->resource->name) && $notification->messageRichParameters->resource->name === $resource) {
			$actualMessage = str_replace(["\r", "\n"], " ", $notification->message);
			$this->deleteNotification($user, [$notification->notification_id]);
		} else {
			Assert::fail("Cannot find notification for resource $resource");
		}
		$expectedMessage = $table->getColumnsHash()[0]['message'];
```

Reply:
you are right this way looks less changes on scenarios. i will implement this